### PR TITLE
Travis script for MacOS build

### PR DESCRIPTION
**Pull request by dcoeurjo**
Update of travis-ci scripts to enable MacOS builds.

:octocat:

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,14 +42,18 @@ matrix:
 
         - os: linux
           compiler: gcc
-          env: GCC_VERSION=5
+        env: GCC_VERSION=5
```
Comment (dictoon) [inline .travis.yml]: Revert change (fix indentation).

File: .travis.yml [hunk 2]
```diff
@@ -22,15 +23,17 @@ addons:
...
 
 install:
   - if [ -n "$GCC_VERSION" ]; then export CC="${CC}-${GCC_VERSION}"; fi
   - if [ -n "$GCC_VERSION" ]; then export CXX="${CXX}-${GCC_VERSION}"; fi
 
 compiler:
+  - clang
   - gcc
 
+
```
Comment (dictoon) [inline .travis.yml]: Revert change.

File: .travis.yml [hunk 3]
```diff
@@ -39,14 +42,18 @@ matrix:
...
         - os: linux
           compiler: gcc
           env: GCC_VERSION=6
+          
+        - os: osx
+          compiler: clang
 
 script:
-  - ./travis/build.sh
+  - if [ $TRAVIS_OS_NAME == linux ]; then ./travis/build.sh ; fi
```
Comment (dictoon) [inline .travis.yml]: Let's rename `build.sh` to `build-linux.sh`?

File: travis/build-macos.sh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+set -e
+
+THISDIR=`pwd`
+mkdir local
+export LD_LIBRARY_PATH=$THISDIR/local/lib:$APPLESEED_DEPENDENCIES/lib:../sandbox/lib/Debug:$LD_LIBRARY_PATH
+
+
+echo "Setting up the deps:"
+echo "------------------------"
```
Comment (dictoon) [inline travis/build-macos.sh]: Adjust separator's length to match title's length.

**Comment by dcoeurjo** [comment]
;)

File: .travis.yml [hunk 5]
```diff
@@ -45,8 +47,20 @@ matrix:
...
+        - compiler: clang
+          os: linux
+
+        - os: linux
+          compiler: gcc
+
+
 script:
-  - ./travis/build.sh
+  - if [ $TRAVIS_OS_NAME == linux ]; then ./travis/build.sh ; fi
```
Comment (dictoon) [inline .travis.yml]: You'll need to fix the path here.

File: .travis.yml [hunk 6]
```diff
@@ -45,8 +47,20 @@ matrix:
           compiler: gcc
           env: GCC_VERSION=6
 
+    exclude:
+        - compiler: gcc
```
Comment (dictoon) [inline .travis.yml]: Keep the same ordering as for includes: os, then compiler.

**Comment by dcoeurjo** [comment]
For the wiki page: 

- `brew install boost-python` is missing (to have boost::python with  python2.7)
- Add ` -DBoost_PYTHON_LIBRARY_RELEASE=/usr/local/lib/libboost_python27.dylib` in the appleseed cmake (I don't know why cmake doest not find it there)

File: .travis.yml (hunk 6 above)
Comment (dcoeurjo) [reply .travis.yml]: 👍

File: .travis.yml (hunk 5 above)
Comment (dcoeurjo) [reply .travis.yml]: pfff... sure ;) sry

**Comment by dictoon** [comment]
> brew install boost-python is missing (to have boost::python with python2.7)

That's interesting. I thought it would be covered by `brew install boost`? Seems like it was on my macOS VM (10.10) since I was able to make the latest build of appleseed without troubles...

> Add -DBoost_PYTHON_LIBRARY_RELEASE=/usr/local/lib/libboost_python27.dylib

Same remark about this one actually... Curious...

**Comment by dcoeurjo** [comment]
For `boost-python`, I didn't know too.. On osx, there is `boost-python` (2.7) and a `boost-python3`... maybe the default `boost` formula just disable python to avoid python version conflicts

**Comment by dcoeurjo** [comment]
https://github.com/Homebrew/homebrew-core/blob/master/Formula/boost.rb#L54

**Comment by dictoon** [comment]
Wow, this is weird... No idea I got my build to have Python then!

**Comment by est77** [comment]
Thanks a lot for doing this, it's really a great improvement. 
We know it is slow and tedious to get it right.

If it makes things easier for you, we have the option to do the same as in linux, setup a repo with some precompiled dependencies for OSX and clone that as part of the travis build.

**Comment by dcoeurjo** [comment]
thanks @est77, debugging travis script/bot with 20/30min latency is always a lot of fun :)...

For the macos pre-built deps, yep that would be very helpful to speedup the process. At this point I first would like to have a passing build and then we'll see.

**Comment by dcoeurjo** [comment]
All green 🎉

**Comment by est77** [comment]
Thanks a lot for this!

**Comment by dictoon** [comment]
Thanks for the terrific work @dcoeurjo, much appreciated!!

**Comment by pjessesco** [comment]
Thanks so much! 
I'll add ` -DBoost_PYTHON_LIBRARY_RELEASE=/usr/local/lib/libboost_python27.dylib` in wiki page.
`boost-python` will be installed when installing `openimageio`..

**Comment by dcoeurjo** [comment]
Great ;). Thx everyone. 

The next step could be to have a deploy on Travis to get the Mac OS binaries for your releases.

**Comment by dictoon** [comment]
> The next step could be to have a deploy on Travis to get the Mac OS binaries for your releases.

Definitely!!

**Pull request merged**